bit/utils.py

```diff
@@ -104,16 +105,21 @@ def get_signatures_from_script(script):
     :returns: A list of retrieved signature from the provided scriptSig.
     :rtype: A ``list`` of ``bytes`` signatures
     """
+
     script = script[1:]  # remove the first OP_0
     sigs = []
-    val, script = read_var_int(script)
-    while val <= 72:  # TODO: Make a better check if the data is a signature (using DER rules: https://bitcoin.stackexchange.com/questions/12554/why-the-signature-is-always-65-13232-bytes-long)
-        if val != 0:  # For partially-signed scriptSigs the missing signatures are each indicated with 0x00 at the end.
-            potential_sig, script = read_bytes(script, val)
-            if bytes_to_hex(potential_sig[0:1]) == '30':
-                sigs.append(potential_sig)
-        if len(script) == 0:  # escape if we have run out of the script
-            break
+    while True:
```

Comment:
I don't like that this is unbounded. Is there a good reason to not put some kind of bounds on it? Maybe it's fine, just a bit eerie.

Reply:
Good point! Especially in this case where this can be transformed to a much more readable and bounded loop! See commit 8f95d52 .